File: Garcia-Bautista-Jonathan_Ulises/Hilos/C++/thread_1.c
#include <stdio.h>
#include <pthread.h>
#include <unistd.h>  

/*
    gcc -o thread_1 thread_1.c
*/

void thread(void);
int main (int argc, char **argv)
{
    int pid;
    pthread_t tid;
    pid = pthread_create(&tid, NULL, (void *)&thread, NULL);
    if( pid < 0 )
    {
        printf("Error en la creacion del Thread\n");
        return 1;
    }
    while(1)
    {
        printf("\tP=%d\n", getpid());
    }
}
void thread(void)
{
    while(1)
    {
        printf("\t\t\t H=%d\n", getpid());
    }
}

File: Garcia-Bautista-Jonathan_Ulises/Hilos/ObjetiveC/thread_1.m
#import <Foundation/Foundation.h>
#import <unistd.h>
#import <sys/types.h>
#import <pthread.h>


void *thread(void *arg);

int main(int argc, const char * argv[]) {
    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
    pthread_t thread_id;
    pthread_create(&thread_id, NULL, thread, NULL);
    while (1) {
        @autoreleasepool {
            NSLog(@"\n\tP = %d", getpid());
        }
        sleep(1);
    }
    [pool release];

    return 0;
}
void *thread(void *arg) {
    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
    while (1) {
        @autoreleasepool {
            NSLog(@"\t\t\t H = %d", getpid());
        }
        sleep(1);
    }
    [pool release];
    return NULL;
}
